daemon/cluster/listen_addr_linux.go

```diff
@@ -0,0 +1,75 @@
+// +build linux
+
+package cluster
+
+import (
+	"net"
+
+	"github.com/vishvananda/netlink"
+)
+
+func (c *Cluster) resolveSystemAddr() (net.IP, error) {
+	// Use the system's only device IP address, or fail if there are
+	// multiple addresses to choose from.
+	interfaces, err := netlink.LinkList()
+	if err != nil {
+		return nil, err
```

Comment:
Would it make sense to fall back to the generic detection function if `LinkList` fails? It looks like Go's `net.Interfaces` function uses netlink on Linux though, so perhaps there's no situation where this will fail and `net.Interfaces` will succeed.